### PR TITLE
Move export prometheus_text_format:escape_label_value outside of TEST def

### PR DESCRIPTION
**Pull request by aaron-seo**
`prometheus_text_format:escape_label_value` is a useful helper function that users can benefit from using without having to duplicate the function in their code.

Namely, it can be used to escape prometheus core metric label values -- based on issue [rabbitmq/rabbitmq-server#9648](https://github.com/rabbitmq/rabbitmq-server/issues/9648) and relevant PR/review from [rabbitmq/rabbitmq-server#9656 ](https://github.com/rabbitmq/rabbitmq-server/pull/9656#discussion_r1355756241)

### CODE REVIEW

File: src/formats/prometheus_text_format.erl
Comment (gomoripeti) [inline src/formats/prometheus_text_format.erl:80]: Existing type-spec is a bit strange, this function does not return `binary()` for input `undefined`

**Review by gomoripeti — APPROVED** [review]

File: src/formats/prometheus_text_format.erl
Comment (ikvmw) [reply src/formats/prometheus_text_format.erl:80]: is it possible to fix in this PR?

**Review by deadtrickster — APPROVED** [review]

File: src/formats/prometheus_text_format.erl
Comment (aaron-seo) [reply src/formats/prometheus_text_format.erl:80]: Looks like my local workspace didn't have GPG signatures setup correctly, so I'll create a new PR anyways with the signed commits. I'll update the spec to the following to more accurately represent the function.
```
-spec escape_label_value(binary() | iolist()) -> binary()
                      ; (undefined) -> no_return().
```

**Comment by michaelklishin** [comment]
@ikvmw you may have to explicitly approve workflow runs for this PR as it is from/by a first-time contributor

**Comment by aaron-seo** [comment]
May need to be merged manually even after approval, due to the job `continuous-integration/travis-ci Expected — Waiting for status to be reported` stalling. @ikvmw @deadtrickster Thanks in advance :)

**Comment by aaron-seo** [comment]
Just bumping this for merge, if there are other tasks needed for the process, please feel free to let me know, thanks @deadtrickster @ikvmw

**Pull request merged**
